### PR TITLE
Adds unit tests for linkedlist

### LINKED ISSUE #36: Unit tests and refactoring
We need 4 unit tests to improve code coverage for a P+. We should also save coverage test reports.

### PR DESCRIPTION
**Pull request by capitanu**
Fixes #36 

Also does some refactoring and adds coverage reports.

Since for a P+ we are doing individual, I also pushed my 2 extra unit tests in here and I think we can add the coverage reports with our names in it in `manual_evaluation/part2_coverage/coverage_results`

### CODE REVIEW

File: manual_evaluations/part2_coverage/coverage_results/linkedlist/calin_test_is_cyclic_after [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Name                                    Stmts   Miss  Cover
+-----------------------------------------------------------
+algorithms/linkedlist/is_cyclic.py         11      3    73%
```
Comment (ArnPellesGit) [inline manual_evaluations/part2_coverage/coverage_results/linkedlist/calin_test_is_cyclic_after:3]: Your cover seems to have gone down here from 80 % -> 73 %. All your other tests imply a greater coverage afterwards. Is this correct?

**Review by ArnPellesGit — CHANGES_REQUESTED** [review]
LGTM Except for one thing: 
Your is_cyclic_before has a coverage of 80%. 
Your is_cyclic_after has a coverage of 73%.

This implies the coverage has decreased which must be a bug or something similar.

File: manual_evaluations/part2_coverage/coverage_results/linkedlist/calin_test_is_cyclic_after (hunk 1 above)
Comment (capitanu) [reply manual_evaluations/part2_coverage/coverage_results/linkedlist/calin_test_is_cyclic_after]: Fixed :D

**Review by ArnPellesGit — APPROVED** [review]
Looks great to me :)

**Pull request merged**